### ISSUE TITLE
Have you tried upstreaming this?

### ISSUE THREAD

**Issue by fasmide** [nmcclain/raspberian-firstboot#5]
I think this would be an awesome addition to the raspberry pi os - have you considered getting this upstreamed?

**Comment by nmcclain** [comment]
I have not, but am fully in support of it! How can I help? I would be happy to make you a co-maintainer if you can drive this process.

**Comment by fasmide** [comment]
Great!

I just finished building the raspberry pi os locally with great success - it's not at all a hard task to accomplish: just pull their repo and run `build-docker.sh` which it takes care of everything. It sets up the build environment and builds the final images, all it needs are a somewhat long time to complete :)

I have a pretty good idea of how to upstream these changes now and I will try to ensure proper credits are given when making the pull request

**Comment by nmcclain** [comment]
Awesome - this would be a really great outcome for this little project!!

Let me know if I can help!

**Comment by beasteers** [comment]
@fasmide is there an upstream issue for this that we could follow?

**Comment by fasmide** [comment]
@beasteers yep right here https://github.com/RPi-Distro/raspberrypi-sys-mods/pull/40

if you got the time - help out testing this stuff out :)

**Comment by nmcclain** [comment]
I will build the May 7th image Monday when I have a Pi I can test on - sorry for the delay.

For the record, the long-term goal for this project is to "go away" just as soon as there is a reasonable first-boot/cloud-init solution in the standard Raspberry Pi OS release.

**Comment by nmcclain** [comment]
I finally built the October 2021 image and released it here.

The AWS S3 bandwidth costs are becoming significant for this project, so I'm hopeful this will get added to the official distro someday sooner than later.

**Comment by fasmide** [comment]
> The AWS S3 bandwidth costs are becoming significant for this project, so I'm hopeful this will get added to the official distro someday sooner than later.

Sounds annoying - I'm sorry but I have no idea how to progress the situation :/

**Comment by samwardbiddle** [comment]
Would love to see how the aws open source folks can support and/or collaborate. This is a great project and would love to see it make it's way into the next release.
